From 5f0abda5e3ca14307bd90f978c0c41e7e08dac57 Mon Sep 17 00:00:00 2001
Message-Id: <5f0abda5e3ca14307bd90f978c0c41e7e08dac57.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Deepak M <m.deepak@intel.com>
Date: Mon, 26 Jan 2015 12:54:43 +0530
Subject: [PATCH 20/95] MUST_REBASE [VPG]: drm/i915: Ignore PIPE interrupts
 during initialization

Ignore interrupts until ADF driver gets initialized.
I915 setup can enable interrupts on all pipes, but it should
be ignored until ADF driver is ready to handle interrupts.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

Issue: GMINL-5544
Change-Id: I346b2a785174d05ebf3b3c486375cd1509e73835
Signed-off-by: Deepak M <m.deepak@intel.com>
---
 drivers/gpu/drm/i915/i915_irq.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_irq.c b/drivers/gpu/drm/i915/i915_irq.c
index 1839088..025ddb1 100644
--- a/drivers/gpu/drm/i915/i915_irq.c
+++ b/drivers/gpu/drm/i915/i915_irq.c
@@ -2340,9 +2340,8 @@ static irqreturn_t cherryview_irq_handler(int irq, void *arg)
 				 * ignore dispaly interrupts for now
 				 */
 
-				/* FIXME: Display interrupts for Pipe C */
 				count = PIPE_A;
-				while (count < PIPE_C) {
+				while (count < I915_MAX_PIPES) {
 					mask = PIPESTAT_IIR(count);
 					pipestat = PIPESTAT(count);
 					pipestat_val = I915_READ(pipestat);
-- 
1.7.9.5

